src/main/java/seedu/lifeasier/storage/FileStorage.java

```diff
@@ -0,0 +1,151 @@
+     * @return True when both files exist.
+     */
+    private boolean fileExists(File saveFileTasks, File saveFileNotes) {
+        return saveFileTasks.exists() && saveFileNotes.exists();
+    }
+
+    /**
+     * Determines if save directory and files are existent, before creating new save files and directories.
+     */
+    private void createNewSaves(File saveFileTasks, File saveFileNotes) {
+        File fileDirectory = new File(DIRECTORY_PATH);
+
+        //If directory exists, create missing save files
+        if (directoryExists(fileDirectory)) {
+            ui.showDirectoryDetected();
+
+            if (!saveFileTasks.exists()) {
+                createTaskSaveFile(saveFileTasks);
+            }
+
+            if (!saveFileNotes.exists()) {
+                createNoteSaveFile(saveFileNotes);
+            }
+
+        } else {
+            handleMissingSaveDirectory(fileDirectory, saveFileTasks, saveFileNotes);
+        }
+    }
+
+    private boolean directoryExists(File fileDirectory) {
+        return fileDirectory.exists();
+    }
+
+    /**
+     * Handles when no save directory is detected and creates a new directory.
+     *
+     * @param fileDirectory File object which holds the directory path to be created.
+     */
+    private void handleMissingSaveDirectory(File fileDirectory, File saveFileTasks, File saveFileNotes) {
+        ui.showNoDirectoryDetected();
+        //Attempt creation of new save directory to hold save files
+        if (createNewDirectory(fileDirectory)) {
+            ui.showNewDirectoryCreated();
+            createTaskSaveFile(saveFileTasks);
+            createNoteSaveFile(saveFileNotes);
+        } else {
+            ui.showNewDirectoryFailCreated();
+        }
+    }
+
+    private boolean createNewDirectory(File fileDirectory) {
+        return fileDirectory.mkdir();
+    }
+
+    /**
+     * Creates a new save file for storing tasks at the specified path.
+     *
+     * @param saveFileTasks File object which holds the path at which the save file is to be created.
+     */
+    private void createTaskSaveFile(File saveFileTasks) {
+        ui.showCreatingNewSaveFile();
+
+        try {
+            if (saveFileTasks.createNewFile()) {
+                ui.showFileCreationSuccess();
+            }
+        } catch (IOException e) {
+            ui.showFileCreationError();
+        }
+    }
+
+    /**
+     * Creates a new save file for storing notes at the specified path.
+     *
+     * @param saveFileNotes File object which holds the path at which the save file is to be created.
+     */
+    private void createNoteSaveFile(File saveFileNotes) {
+        ui.showCreatingNewNotesSaveFile();
+
+        try {
+            if (saveFileNotes.createNewFile()) {
+                ui.showFileCreationSuccess();
+            }
+        } catch (IOException e) {
+            ui.showFileCreationError();
+        }
+    }
```

Comment:
I think these functions only differ in the parameter you are passing in. You could consider making it one function?

Reply:
Alright, ive tried my best to refactor the function into a single method